### PR TITLE
Add net.gpro.gproapp

### PR DESCRIPTION
**Pull request by gpronet**
GPRO is a long term cross platform racing management game where you get to control and influence a racing team of your own and compete against thousands of managers from around the world.
 
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: net.gpro.gproapp.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      - install -D /app/share/icons/hicolor/128x128/apps/gpro.png /app/share/icons/hicolor/128x128/apps/net.gpro.gproapp.png
+      
+      - rm /app/share/icons/hicolor/16x16/apps/gpro.png
+      - rm /app/share/icons/hicolor/32x32/apps/gpro.png 
+      - rm /app/share/icons/hicolor/48x48/apps/gpro.png 
+      - rm /app/share/icons/hicolor/64x64/apps/gpro.png 
+      - rm /app/share/icons/hicolor/128x128/apps/gpro.png
+      - rm /app/share/icons/hicolor/256x256/apps/gpro.png
+      - rm /app/share/icons/hicolor/512x512/apps/gpro.png
+      - rm /app/share/icons/hicolor/1024x1024/apps/gpro.png
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:67]: use `rename-icon` instead in the manifest.

File: net.gpro.gproapp.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+      # Install app wrapper
+      
+      #- install -Dm755 gpro-entrypoint.sh /app/bin/gpro
+      - find /app/GPRO/ -exec chmod -R a-s,go+rX,go-w {} \;
+      - sed -i 's/Game\,SportsGame\,StrategyGame\,Simulation\,RolePlaying/Game/g' /app/share/applications/gpro.desktop
+      - mv /app/share/applications/gpro.desktop /app/share/applications/net.gpro.gproapp.desktop
+      - desktop-file-edit --set-key="Exec" --set-value="startgpro" --set-icon=$FLATPAK_ID /app/share/applications/net.gpro.gproapp.desktop --set-key="Categories" --set-value="Game;StrategyGame;"
+      - install -Dm644 net.gpro.gproapp.metainfo.xml /app/share/metainfo/net.gpro.gproapp.metainfo.xml
+      - install -Dm644 net.gpro.gproapp.metainfo.xml /app/share/metainfo/net.gpro.gproapp.appdata.xml
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:47]:
```suggestion
      - install -Dm644 net.gpro.gproapp.metainfo.xml /app/share/metainfo/net.gpro.gproapp.metainfo.xml
```

File: net.gpro.gproapp.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      - tar xf data.tar.xz -C /app --strip-components=2 ./opt ./usr
+      - rm -f control.tar.gz data.tar.xz debian-binary gpro.deb
+
+      # Install app wrapper
+      
+      #- install -Dm755 gpro-entrypoint.sh /app/bin/gpro
+      - find /app/GPRO/ -exec chmod -R a-s,go+rX,go-w {} \;
+      - sed -i 's/Game\,SportsGame\,StrategyGame\,Simulation\,RolePlaying/Game/g' /app/share/applications/gpro.desktop
+      - mv /app/share/applications/gpro.desktop /app/share/applications/net.gpro.gproapp.desktop
+      - desktop-file-edit --set-key="Exec" --set-value="startgpro" --set-icon=$FLATPAK_ID /app/share/applications/net.gpro.gproapp.desktop --set-key="Categories" --set-value="Game;StrategyGame;"
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:45]: why do you need to edit it since you have included the file in this repo.

File: net.gpro.gproapp.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      - ar x gpro.deb
+      - tar xf data.tar.xz -C /app --strip-components=2 ./opt ./usr
+      - rm -f control.tar.gz data.tar.xz debian-binary gpro.deb
+
+      # Install app wrapper
+      
+      #- install -Dm755 gpro-entrypoint.sh /app/bin/gpro
+      - find /app/GPRO/ -exec chmod -R a-s,go+rX,go-w {} \;
+      - sed -i 's/Game\,SportsGame\,StrategyGame\,Simulation\,RolePlaying/Game/g' /app/share/applications/gpro.desktop
+      - mv /app/share/applications/gpro.desktop /app/share/applications/net.gpro.gproapp.desktop
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:44]: `rename-desktop-file` should do the trick.

File: net.gpro.gproapp.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    build-commands:
+      - ar x gpro.deb
+      - tar xf data.tar.xz -C /app --strip-components=2 ./opt ./usr
+      - rm -f control.tar.gz data.tar.xz debian-binary gpro.deb
+
+      # Install app wrapper
+      
+      #- install -Dm755 gpro-entrypoint.sh /app/bin/gpro
+      - find /app/GPRO/ -exec chmod -R a-s,go+rX,go-w {} \;
+      - sed -i 's/Game\,SportsGame\,StrategyGame\,Simulation\,RolePlaying/Game/g' /app/share/applications/gpro.desktop
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:43]: don't use sed. use desktop file edit.

File: net.gpro.gproapp.desktop
Comment (hfiguiere) [inline net.gpro.gproapp.desktop:1]: why is this there seems to be one from upstream according to the build steps?

File: startgpro [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+export TMPDIR="$XDG_RUNTIME_DIR/app/${FLATPAK_ID:-net.gpro.gproapp}"
+
+declare -a FLAGS=(--ozone-platform-hint=auto)
+
+if [[ $XDG_SESSION_TYPE == "wayland" ]] && [[ -c /dev/nvidia0 ]]
```
Comment (hfiguiere) [inline startgpro:7]: why does this exist anyway you haven't enabled wayland.

File: startgpro [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+then
+    echo "Using NVIDIA on Wayland, disabling gpu sandbox"
+    FLAGS+=(--disable-gpu-sandbox)
+fi
+
+echo "Passing the following arguments to Electron:" "${FLAGS[@]}"
+zypak-wrapper /app/GPRO/gpro flatpak=1 --no-sandbox "${FLAGS[@]}" "$@"
```
Comment (hfiguiere) [inline startgpro:14]:
```suggestion
exec zypak-wrapper /app/GPRO/gpro flatpak=1 --no-sandbox "${FLAGS[@]}" "$@"
```

File: net.gpro.gproapp.yml [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        path: net.gpro.gproapp.svg
+      - type: file
+        path: net.gpro.gproapp.metainfo.xml
+      - type: file
+        path: startgpro
+    #  - type: script
+     #   dest-filename: gpro-entrypoint.sh
+      #  commands:
+       #   - export TMPDIR="$XDG_RUNTIME_DIR/app/${FLATPAK_ID}"
+        #  - exec zypak-wrapper /app/GPRO/gpro flatpak=1 --no-sandbox "$@"
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:94]:
```suggestion
```

File: net.gpro.gproapp.yml (hunk 1 above)
Comment (hfiguiere) [reply net.gpro.gproapp.yml:67]: still remove the 1024x1024 thought, it's an invalid size

File: net.gpro.gproapp.yml [hunk 9]
```diff
@@ -0,0 +1,94 @@
+app-id: net.gpro.gproapp
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
```
Comment (hfiguiere) [inline net.gpro.gproapp.yml:8]: why do you need this since you are not building from source?

(or better why aren't you building from source?)

**Review by hfiguiere — CHANGES_REQUESTED** [review]
questions haven't been answered.

File: net.gpro.gproapp.svg
Comment (hfiguiere) [inline net.gpro.gproapp.svg:1]: it already comes with icons. why adding this here?

File: net.gpro.gproapp.desktop
Comment (hfiguiere) [inline net.gpro.gproapp.desktop:1]: you mean the debian package doesn't ship a desktop file? this is a bug. It should and should be used here instead of adding one to the repo.

Comment (hfiguiere) [reply net.gpro.gproapp.desktop:1]: see https://github.com/flathub/flathub/pull/5358#discussion_r1655377458 when I previously asked.

File: net.gpro.gproapp.yml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+rename-icon: gpro
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  # required for notifications in various desktop environments
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:19]:
```suggestion
```

Notifications go through the portal by default

File: net.gpro.gproapp.yml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/gpro/flatpak-node/cache
+        npm_config_cache: /run/build/gpro/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+    build-commands:
+      - ar x gpro.deb
+      - tar xf data.tar.xz -C /app --strip-components=2 ./opt ./usr
+      - rm -f control.tar.gz data.tar.xz debian-binary gpro.deb
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:38]:
```suggestion
```

these aren't being installed.

File: net.gpro.gproapp.yml [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+      - find /app/GPRO/ -exec chmod -R a-s,go+rX,go-w {} \;
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:40]: why is this chmod operating on everything?

File: startgpro [hunk 13]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+export TMPDIR="$XDG_RUNTIME_DIR/app/${FLATPAK_ID:-net.gpro.gproapp}"
+
+declare -a FLAGS=(--ozone-platform-hint=auto)
+
+echo "Passing the following arguments to Electron:" "${FLAGS[@]}"
+exec zypak-wrapper /app/GPRO/gpro flatpak=1 --no-sandbox "${FLAGS[@]}" "$@"
```
Comment (bbhtt) [inline startgpro:8]: zypak-wrapper means you don't need to disable the inbuild sandbox.

File: net.gpro.gproapp.metainfo.xml
Comment (bbhtt) [inline net.gpro.gproapp.metainfo.xml:1]: this file should come from upstream

File: net.gpro.gproapp.yml [hunk 14]
```diff
@@ -0,0 +1,58 @@
+app-id: net.gpro.gproapp
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:6]:
```suggestion
runtime-version: '24.08'
sdk: org.freedesktop.Sdk
base: org.electronjs.Electron2.BaseApp
base-version: '24.08'
```

File: net.gpro.gproapp.yml [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: gpro
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/gpro/flatpak-node/cache
+        npm_config_cache: /run/build/gpro/flatpak-node/npm-cache
+        npm_config_offline: 'true'
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:30]: why is this needed ? You are reusing debs.

File: net.gpro.gproapp.yml [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --talk-name=org.kde.StatusNotifierWatcher
+  # needed to integrate with Steam for profile linking
+  - --filesystem=home
+build-options:
+  append-path: /usr/lib/sdk/node18/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:22]: same here. You are reusing debs.

**Comment by bbhtt** [comment]
bot, build net.gpro.gproapp

**Comment by bbhtt** [comment]
bot, build net.gpro.gproapp

File: net.gpro.gproapp.yml [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '24.08'
+command: startgpro
+rename-icon: gpro
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
```
Comment (bbhtt) [inline net.gpro.gproapp.yml:12]: Doesn't work on wayland

```
assing the following arguments to Electron: --ozone-platform-hint=auto
[2:1128/072955.981543:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
runCount2
stesmworks,  {
  init: [Function (anonymous)],
  restartAppIfNecessary: [Function (anonymous)],
  electronEnableSteamOverlay: [Function (anonymous)],
  SteamCallback: {
    PersonaStateChange: 0,
    SteamServersConnected: 1,
    SteamServersDisconnected: 2,
    SteamServerConnectFailure: 3,
    LobbyDataUpdate: 4,
    LobbyChatUpdate: 5,
    P2PSessionRequest: 6,
    P2PSessionConnectFail: 7
  }
}
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
dlopen failed trying to load:
/home/test/.steam/sdk64/steamclient.so
with error:
/home/test/.steam/sdk64/steamclient.so: cannot open shared object file: No such file or directory
[S_API] SteamAPI_Init(): Sys_LoadModule failed to load: /home/test/.steam/sdk64/steamclient.so
[2:1128/072956.235500:ERROR:wayland_connection.cc(198)] Failed to connect to Wayland display: No such file or directory (2)
[2:1128/072956.235524:ERROR:ozone_platform_wayland.cc(264)] Failed to initialize Wayland platform
[2:1128/072956.235528:ERROR:env.cc(258)] The platform failed to initialize.  Exiting.
```

Comment (bbhtt) [reply net.gpro.gproapp.yml:12]: If you want to experiment with `--ozone-platform-hint=auto` use fallback-x11+wayland sockets, otherwise don't pass that and follow the Electron default (keep x11 only)

**Comment by bbhtt** [comment]
bot, build net.gpro.gproapp
